From: Gerrit Pape <pape@smarden.org>
Date: Sat, 9 Dec 2006 10:32:01 +0000
Subject: tryto.c

Don't report failure if the final try of prog succeeds (thx Andras
Korn, closes: #401547).
---
 src/tryto.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tryto.c b/src/tryto.c
index 7d42315..94c0d01 100644
--- a/src/tryto.c
+++ b/src/tryto.c
@@ -237,7 +237,7 @@ int main (int argc, const char * const *argv, const char * const *envp) {
       }
       break;
     }
-    if (rc == 0) break;
+    if (rc == 0) _exit(0);
     if (verbose) strerr_warn2(WARNING, "child crashed.", 0);
     if (lseek(0, 0, SEEK_SET) != 0)
 	if (verbose) strerr_warn2(WARNING,
@@ -246,7 +246,7 @@ int main (int argc, const char * const *argv, const char * const *envp) {
     sleep(1);
   }
 
-  if (processor && (rc != 0)) {
+  if (processor) {
     for (;;) {
       int r;
       char *s;
